lesson21/main.js

```diff
@@ -119,23 +124,92 @@ const formingTableHeaderNameWithKey = (key) => {
 const createTableData = (usersData) => {
   const fragment = document.createDocumentFragment();
   usersData.forEach((userData) => {
-    const trOfTdata = document.createElement("tr");
+    const trOfTdata = createElementWithClassName("tr", "users-table__tr-td");
     /**
      *Extract value from each of usersData
      * @param {Object} userData the object of each users data
      * @param {String,Number} val value of each users data
      */
 
-    Object.values(userData).forEach((val) => {
+    Object.keys(userData).forEach((key) => {
+      if (key === "id") return;
       const td = createElementWithClassName("td", "users-table__td");
-      td.textContent = val;
+      td.textContent = userData[key];
       trOfTdata.appendChild(td);
     });
     fragment.appendChild(trOfTdata);
   });
   return fragment;
 };
 
+const createSortBtn = () => {
+  const btn = createElementWithClassName("button", "sort-btn");
+  const sortArrow = createElementWithClassName("img", "sort-img");
+  // bothのイメージをデフォルトとする
+  sortArrow.src = "./img/both.svg";
+  btn.appendChild(sortArrow);
+  return btn;
+};
+
+const renderSortBtn = () => {
+  const btn = createSortBtn();
+  const targetEle = document.getElementById("id");
+  targetEle.appendChild(btn);
+};
+
+const clickSortBtn = (usersData) => {
+  const sortArrow = document.querySelector(".sort-btn");
+  const sortArrowImg = document.querySelector(".sort-img");
+  sortArrow.addEventListener("click", () => {
+    changeSortStateAndArrowImg(sortArrowImg);
+    rerenderTableData(usersData);
+  });
+};
+
+const changeSortStateAndArrowImg = (sortArrow) => {
+  if (sortState === "both") {
+    sortState = "asc";
+    sortArrow.src = "/img/asc.svg";
+  } else if (sortState === "asc") {
+    sortState = "desc";
+    sortArrow.src = "/img/desc.svg";
+  } else {
+    sortState = "both";
+    sortArrow.src = "/img/both.svg";
+  }
+};
+
+const rerenderTableData = (usersData) => {
+  const table = document.getElementById("js-table");
+  const copiedUsersData = [...usersData];
+  if (sortState === "asc") {
+    sortAsc(copiedUsersData);
+  } else if (sortState === "desc") {
+    sortDesc(copiedUsersData);
+  }
```

Comment:
IMO
Here, too!
```js
  if (sortState === "asc") sortAsc(copiedUsersData);
  if (sortState === "desc") sortDesc(copiedUsersData);
```

Reply:
I also missed this part, thank you for teaching me!
And also I changed implementation of `rerenderTableData` and each `sort` functions as Mr.Morita taught me in Discord!
[f74c4cd](https://github.com/yuka830/js-lessons/pull/27/commits/f74c4cdf810efdf6004773d02a4498d30e82d33c)